gulp e2e failure with IE browser

NOTE : Windows VMtempalate install IE11 by default while i am working to find steps to downgrade it to IE10 , i am seeing below error runing gulp e2e command below : 

Pre req : ML 8.0.1.1 ;  Win 7 ;  Develop branch 


gulp e2e --middle-tier=external --browser=ie 

![ie11](https://cloud.githubusercontent.com/assets/6935682/6973026/963af9fc-d93c-11e4-83d5-9a49d7ff07bf.jpg)

```
        at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_mod
        at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protract
        at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules
        at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protracto
        at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-sampl
        at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\pr
        at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)

    And I attempt to log in with invalid credentials # specs\features\login-logout\login.feat
    And my login attempt is denied                   # specs\features\login-logout\login.feat
    And I attempt to log in as a Contributor         # specs\features\login-logout\login.feat
    Then I am logged in                              # specs\features\login-logout\login.feat


  Scenario: Log In as Mary           # specs\features\login-logout\login.feature:17
Possibly unhandled TypeError: undefined is not a function
    at C:\Users\builder\marklogic-samplestack\node_modules\connect-marklogic\lib\connect-mark
    at tryCatch1 (C:\Users\builder\marklogic-samplestack\node_modules\marklogic\node_modules\
    at Promise$_callHandler [as _callHandler] (C:\Users\builder\marklogic-samplestack\node_mo
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (C:\Users\builder\mar
    at Promise$_settlePromiseAt [as _settlePromiseAt] (C:\Users\builder\marklogic-samplestack
    at Promise$_settlePromises [as _settlePromises] (C:\Users\builder\marklogic-samplestack\n
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (C:\Users\builder\marklogic-s
    at Async$consumeFunctionBuffer (C:\Users\builder\marklogic-samplestack\node_modules\markl
    at process._tickCallback (node.js:419:13)
    Given I am "Mary"                # specs\features\login-logout\login.feature:18
      NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open"
        at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_mod
        at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protract
        at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules
        at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protracto
        at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-sampl
        at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\pr
        at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)

    Then my user name is "MaryAdmin" # specs\features\login-logout\login.feature:19

@view-contributor
Feature: View Contibutor

  Any user may view "contributor detail" about themselves or other contributors
  of content to the system. If the user is a "local contributor", a custom
  dialog displays basic information. If the user is a StackOverflow user, then
  a new window is opened to display that user's StackOverflow profile.


  Scenario: View Contributor Mary                                                # specs\feat
Possibly unhandled TypeError: undefined is not a function
    at C:\Users\builder\marklogic-samplestack\node_modules\connect-marklogic\lib\connect-mark
    at tryCatch1 (C:\Users\builder\marklogic-samplestack\node_modules\marklogic\node_modules\
    at Promise$_callHandler [as _callHandler] (C:\Users\builder\marklogic-samplestack\node_mo
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (C:\Users\builder\mar
    at Promise$_settlePromiseAt [as _settlePromiseAt] (C:\Users\builder\marklogic-samplestack
    at Promise$_settlePromises [as _settlePromises] (C:\Users\builder\marklogic-samplestack\n
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (C:\Users\builder\marklogic-s
    at Async$consumeFunctionBuffer (C:\Users\builder\marklogic-samplestack\node_modules\markl
    at process._tickCallback (node.js:419:13)
    Given I am "Mary"                                                            # specs\feat
      NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open"
        at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_mod
        at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protract
        at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules
        at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protracto
        at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-sampl
        at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\pr
        at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)

    And I am using the brief seed data                                           # specs\feat
    And I visit the "qnadoc" page with id "5dce8909-0972-4289-93cd-f2e8790a17fc" # specs\feat
    And I focus on the question                                                  # specs\feat
    And I view the content contributor                                           # specs\feat
    Then the contributor display name is "MaryAdmin"                             # specs\feat
    And the contributor votes cast are greater than "4"                          # specs\feat
    And the contributor reputation is greater than "99"                          # specs\feat
    And dismiss the dialog                                                       # specs\feat


  Scenario: View Contributor Joe                                                 # specs\feat
Possibly unhandled TypeError: undefined is not a function
    at C:\Users\builder\marklogic-samplestack\node_modules\connect-marklogic\lib\connect-mark
    at tryCatch1 (C:\Users\builder\marklogic-samplestack\node_modules\marklogic\node_modules\
    at Promise$_callHandler [as _callHandler] (C:\Users\builder\marklogic-samplestack\node_mo
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (C:\Users\builder\mar
    at Promise$_settlePromiseAt [as _settlePromiseAt] (C:\Users\builder\marklogic-samplestack
    at Promise$_settlePromises [as _settlePromises] (C:\Users\builder\marklogic-samplestack\n
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (C:\Users\builder\marklogic-s
    at Async$consumeFunctionBuffer (C:\Users\builder\marklogic-samplestack\node_modules\markl
    at process._tickCallback (node.js:419:13)
    Given I am "Joe"                                                             # specs\feat
      NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open"
        at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_mod
        at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
        at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\seleni
        at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\mar
)
        at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_module
        at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protract
        at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules
        at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protracto
        at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-sampl
        at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\pr
        at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)

    And I am using the brief seed data                                           # specs\feat
    And I visit the "qnadoc" page with id "778d0b9c-419f-496a-a300-44815d79708d" # specs\feat
    And I focus on the question                                                  # specs\feat
    And I view the content contributor                                           # specs\feat
    Then the contributor display name is "JoeUser"                               # specs\feat
    And the contributor votes cast are greater than "2"                          # specs\feat
    And the contributor reputation is greater than "49"                          # specs\feat
    And dismiss the dialog                                                       # specs\feat


(::) failed steps (::)

NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open")
  at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:15
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\s
  at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protractor\nod
  at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules\protr
  at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
  at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-samplestack
  at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\protract
  at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)


NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open")
  at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:15
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\s
  at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protractor\nod
  at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules\protr
  at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
  at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-samplestack
  at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\protract
  at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)


AssertionError: expected 54 to equal 77



NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open")
  at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:15
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\s
  at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protractor\nod
  at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules\protr
  at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
  at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-samplestack
  at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\protract
  at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)


NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open")
  at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:15
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\s
  at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protractor\nod
  at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules\protr
  at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
  at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-samplestack
  at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\protract
  at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)


NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open")
  at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:15
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\s
  at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protractor\nod
  at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules\protr
  at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
  at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-samplestack
  at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\protract
  at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)


NoSuchElementError: No element found using locator: By.className("ss-login-dialog-open")
  at new bot.Error (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modul
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\lib\element.js:673:15
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at notifyAll (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\s
  at resolve (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at fulfill (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sel
  at C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\selenium-web
  at [object Object].webdriver.promise.ControlFlow.runInNewFrame_ (C:\Users\builder\marklogic
  at notify (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node_modules\sele
  at [object Object].then (C:\Users\builder\marklogic-samplestack\node_modules\protractor\nod
  at Object.webdriver.promise.asap (C:\Users\builder\marklogic-samplestack\node_modules\protr
  at newFrame.onComplete (C:\Users\builder\marklogic-samplestack\node_modules\protractor\node
  at [object Object].webdriver.promise.Frame_.notify_ (C:\Users\builder\marklogic-samplestack
  at [object Object]._onTimeout (C:\Users\builder\marklogic-samplestack\node_modules\protract
  at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)


Failing scenarios:
C:\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.featur
C:\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:7 # Scenar
C:\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:18 # Scena
C:\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:8 # Scenario
C:\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:17 # Scenari
C:\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:9 # Scenario:
C:\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:20 # Scenario:

11 scenarios (7 failed, 4 passed)
92 steps (7 failed, 54 skipped, 31 passed)
Shutting down selenium standalone server.
[launcher] 0 instance(s) of WebDriver still running
[launcher] internet explorer #1 failed 7 test(s)
[launcher] overall: 7 failed spec(s)
[launcher] Process exited with error code 1
```

--- a/appserver/java-spring/static/_marklogic/services/data/mlModelBase.js
+++ b/appserver/java-spring/static/_marklogic/services/data/mlModelBase.js
@@ -175,10 +175,7 @@ define(['_marklogic/module'], function (module) {
       };
 
       MlModel.prototype.onResponseDELETE = function (data) {
-        // we normally consider it an error if delete has a body
-        if (data && Object.keys(data).length) {
-          throw new Error('don\'nt assign data on DELETE');
-        }
+        // do nothing with responses to DELETE;
       };
 
       /**

--- a/browser/src/_marklogic/services/data/mlModelBase.js
+++ b/browser/src/_marklogic/services/data/mlModelBase.js
@@ -175,10 +175,7 @@ define(['_marklogic/module'], function (module) {
       };
 
       MlModel.prototype.onResponseDELETE = function (data) {
-        // we normally consider it an error if delete has a body
-        if (data && Object.keys(data).length) {
-          throw new Error('don\'nt assign data on DELETE');
-        }
+        // do nothing with responses to DELETE;
       };
 
       /**

--- a/browser/test/unit-tests/_marklogic/services/data/mlModelBase.unit.js
+++ b/browser/test/unit-tests/_marklogic/services/data/mlModelBase.unit.js
@@ -154,24 +154,7 @@ define(['testHelper'], function (helper) {
       });
 
       it(
-        'should throw an error if someone tries to do anything on DELETE',
-        function () {
-          var instance = impl1.create({
-            id: '1'
-          });
-          try {
-            instance.onHttpResponse({ stuff: 'bad'}, 'DELETE');
-          }
-          catch (err) {
-            assert(true);
-            return;
-          }
-          assert(false, 'no error was thrown');
-        }
-      );
-
-      it(
-        'should NOT throw an error if someone tries to do nothing on DELETE',
+        'should not error on DELETE',
         function () {
           var instance = impl1.create({
             id: '1'

